### PR TITLE
SDMEXT-546 (Setup PR merge with cf push) 

### PR DESCRIPTION
**Pull request by vibhutikumar07**
Jira: [SDMEXT-546](https://jira.tools.sap/browse/SDMEXT-546)

The workflow will be triggered whenever a pull request is merged into the develop branch.

Tested the workflow on one of the local space : https://github.com/cap-java/sdm/actions/runs/12233005419

### CODE REVIEW

**Comment by ashishjain14** [comment]
Hi @vibhutikumar07 : Looks fine, are we also planning to add integration test run with this PR?

File: .github/workflows/cfdeploy.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          wget -P /tmp https://github.com/SAP/cloud-mta-build-tool/releases/download/v1.2.28/cloud-mta-build-tool_1.2.28_Linux_amd64.tar.gz
+          tar -xvzf /tmp/cloud-mta-build-tool_1.2.28_Linux_amd64.tar.gz
+          sudo mv mbt /usr/local/bin/
+          
+          # Build the MTA project
+          echo "Running mbt build"
+          mbt build -p=cf -t mta_archives --mtar demoappjava-1.0.0.mtar
+          
+          # Install cf CLI plugin
+          cf install-plugin multiapps -f
```
Comment (yashmeet29) [inline .github/workflows/cfdeploy.yml:72]: Why are we installing this multiple times? In Login to Cloud Foundry step also we installed this.

File: .github/workflows/cfdeploy.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+      - name: Build and package
+        run: |
+          cd sdm
+          mvn clean install
+
+      - name: Verify and Checkout Deploy Branch
+        run: |
+          git fetch origin
+          if git rev-parse --verify origin/deploy; then
```
Comment (yashmeet29) [inline .github/workflows/cfdeploy.yml]: Shouldn't this be ```if git rev-parse --verify origin/develop_deploy; then```

**Review by yashmeet29 — APPROVED** [review]

**Comment by vibhutikumar07** [comment]
> Hi @vibhutikumar07 : Looks fine, are we also planning to add integration test run with this PR?

Hi @ashishjain14 : As of now, the integration job has not been completed. Once it is finalized, we will proceed with the integration.

**Pull request merged**